source/data/workflow_automation.rst

```diff
@@ -0,0 +1,231 @@
+.. _workflow_automation.rst:
+
+Workflow automation
+===================
+
+iRODS provides a Rule System to automate data management tasks. Each organisation or project has its own policies and needs with regards to file housekeeping and metadata extraction. Examples of file operations are making regular backups, checking file integrity, cleaning up permissions, emptying the trash,... Metadata can be extracted from data files or accompanying text files to be stored to the iCAT catalog. This frees human users of having to apply repetitive actions, and ensures policies are applied consistently and without error.
+
+This is made possible by the Rule Engine Plugin Framework (REPF), which executes the rules and keeps track of the execution state of all active rules (as rules can be immediate, have a delay or a condition). The framework keeps track of both User-level rules and System-level rules. 
+
+User-level rules are stored locally and manually invoked by any user using the ``irule`` command, which runs it in the iRODS server. They are meant for personal or group use (if the user is part of that group) and are typically simple 'one-shot' workflow operations. System-defined rules are stored server-side by an iRODS developer or administrator and automatically invoked by the Rule Engine when a certain condition is met. This condition is called an 'event' or more formally a Policy Enforcement Point (PEP). They are meant for consistent data management of a whole zone or for complex group/project data management tasks.
+
+Rules can be invoked in three ways: by directly calling the ``irule`` command on a rule file (only for User-level rules); by reaching a Policy Enforcement Point, triggering a System-level rule; or periodically via delayed execution (both for User-level and System-level rules). The delay mechanism is used for resource-heavy, time-intensive processes that can occur one or more times in the future at regular intervals.
+
+The Python iRODS Client (PRC) is executed client-side, making it somewhat less efficient than iRODS rule execution. Although both offer overlapping functionality, the delay mechanism used by iRODS rules is more graceful and these rules get stored centrally in the REPF. It is also possible to invoke a rule via the PRC.
+
+This article focuses only on User-level rules. If you have more complex data processing pipelines, the Tier-1 Data Team (FOZ-RDM) at KU Leuven can create System-level rules for you.
+
+Rule syntax
+-----------
+
+User-level rules can only be written in the iRODS Rule language, System-level rules can also be written in other languages (Python, C++ or JavaScript). The iRODS Rule language is a domain specific scripting language composed of simple building blocks. It uses curly brackets, # for comments, variable names start with '*' and strings are enclosed by ' or ". The documentation for the language can be found here: https://docs.irods.org/4.2.10/plugins/irods_rule_language/.
```

Comment:
@mstfdkmn can user rules also be written in Python?

Reply:
Mustafa explained that technically they can but due to a limitation (bug) they are not available for user rules. If or how we express this I leave to you: 'In general they are, but due to technical limitations in our HPC zone they aren't ' or something like this.